### DISCUSSION TITLE
Meaning of brute force detected parameters 10328 and 10329

### DISCUSSION THREAD

**Discussion by fredlcore** [fredlcore/BSB-LAN#310] [Q&A] [answered]
Parameters 10328 and 10329 (command IDs 0x0D3D0944 and 0x0D3D0945 respectively) seem to be temperatures that define between which values the boiler flowtemp setpoints can vary. Here are some observations:
```

2210 Kessel - Sollwert Minimum            = 40 °C (constant)
2211 Kessel - Sollwert Minimum OEM        = 40 °C (constant)
2212 Kessel - Sollwert Maximum            = 80 °C (constant)
2213 Not supported
2214 Not supported

10328 Unknown name (minimum)                  = 59.3 °C (changing constantly, read only)
10329 Unknown name (maximum)                  = 51.3 °C (changing constantly, read only)

8310 Boiler temp actual value                 = 58.6 °C
8311 Boiler temp setpoint: °C (parameter not supported)
8743 Flow temp actual value heat circuit 1:   = 31.4 °C
8744 Flow temperature setpoint H1:            = 31.0 °C

So the formula is the following:

v(10328) = v(8744) + v(830) + v(2240) / 2
v(10329) = v(8744) + v(830) - v(2240) / 2
TKcurrent-max = target flow temperature + boost + switching diff / 2
TKcurrent-min = target flow temperature + boost - switching diff / 2

See previous:
TV: target flow temperature = 30.3°C (command 8743)
B: Boost of the flow temperature setpoint mixing valve = 25°C (command 830 Mischerüberhöhung) 
TKw: Boiler temperature setpoint = TV + B = 30.3°C + 25°C = 55.3°C
SDK: Switching differential = 8°C (command 2240 Schaltdifferenz Kessel)
TKcurrent-max = TKw + SDK/2 = 55.3°C + 8°C / 2 = 59.3°C (correct, see the measurement)
TKcurrent-min = TKw - SDK/2 = 51.3°C - 8°C / 2 = 51.3°C (correct, see the measurement)
```

The questions now are: How should these two parameters be named correctly? And what parameter numbers should these be assigned to? It seems that these parameters are not "official" in any way, yet may be quite helpful for analyzing the heating system's behaviour.

**Comment by 1coderookie** [comment]
I think I'm on the right path, also with corresponidng parameters which already exists, but before I talk about that I need to do some more tests though..

Btw: I _think_ the naming above 
```
10328 Unknown name (minimum)                  = 59.3 °C (changing constantly, read only)
10329 Unknown name (maximum)                  = 51.3 °C (changing constantly, read only)
```
is wrong anyway, cuz the 28 is the max and the 29 is the min based on that values and the calculation below. 
But as I said, I think I probably got it anyway..

**Comment by henkosch** [comment]
Yeah sorry, they are the other way around:

```
10328 Unknown name (minimum)                  = 51.3 °C (changing constantly, read only)
10329 Unknown name (maximum)                  = 59.3 °C (changing constantly, read only)
```

And halfway between the two is the boiler setpoint temperature.

**Comment by henkosch** [comment]
Here are some of my measurements on a chart:

1. chart shows 8310, 10328, 10329 (note that 10328 and 10329 are changing together to lower values as outside temperature rises)
2. chart shows 8743, 8744 (flow setpoint is also dropping as outside temperature rises)
3. chart shows outside temperature

<img width="1638" alt="Screenshot 2021-01-25 at 10 51 29" src="https://user-images.githubusercontent.com/908091/105690041-d3507600-5efb-11eb-9e01-7e32262c82ad.png">
<img width="1638" alt="Screenshot 2021-01-25 at 10 52 56" src="https://user-images.githubusercontent.com/908091/105690055-d64b6680-5efb-11eb-8c4b-5c63683cd516.png">
<img width="1638" alt="Screenshot 2021-01-25 at 10 53 53" src="https://user-images.githubusercontent.com/908091/105690079-da778400-5efb-11eb-94ae-c89d79e731dd.png">

**Comment by henkosch** [comment]
Here is a longer interval (24h) of the boiler temperatures and setpoints:

You can see the peeks where the DHW overrides the boiler setpoint.

<img width="1638" alt="Screenshot 2021-01-25 at 11 01 32" src="https://user-images.githubusercontent.com/908091/105690834-c1bb9e00-5efc-11eb-9ed8-4260bfb5d110.png">

**Comment by henkosch** [comment, marked as answer]
Here is what the documentation calls these values:

10328 : Setpoint for switching on
10329 : Setpoint for switching off

![Képernyőfotó 2021-01-24 - 18 45 38](https://user-images.githubusercontent.com/908091/105692103-4e1a9080-5efe-11eb-9de1-8c0b255ec5b6.png)

**Comment by 1coderookie** [comment]
Thats good because you say it fits your calculation.
It's interesting though that it doesn't seem to fit my test results and that it even seems to differ between a RVS43 and a RVS63 (which usually couldn't/shouldn't be cuz the RVS63 is just a bigger RVS43), but I'm having a hard time to concentrate right now (my cat got intoxicated with rat poison), sorry, so I think I'll quit for today - I don't want to mix anything up in my mind and present wrong conclusions. But I'll also look up the manual for that RVS43/63 and try if I find something, also I'll continue with testing around.. Just for your information in case nothing comes anymore today from side..

**Comment by fredlcore** [comment]
Any news here?

**Comment by 1coderookie** [comment]
Oopps, shame on me - didn't get into it much further since that day.. :(

**Comment by fredlcore** [comment]
Should we continue this at some point?

**Comment by 1coderookie** [comment]
Sure, but until now I didn't came to any new conclusion what these parameters are at RVS43&RVS63 controllers. As I wrote before, my findings and the description from the manual above don't really fit with these controller types, and I didn't find anything new within the manuals yet. But I have written myself a note which is pinned at my desktop here, so when I have more time for that, I'll keep on searching. 
Until then: If you want to implement these parameters, I'd suggest to implement them with the numbers and names as @henkosch mentioned based on his findings and the manual, but probably only for that type of PPS controller.

**Comment by fredlcore** [comment]
I'm digging this up again because I also didn't have time to follow-up on this, but if we take @henkosch's findings and name the parameters accordingly, where should we put these? I would try to move all identified parameters out of the 10000+ range into properly fitting categories because otherwise people won't find them and also the term "Setpoint for switching on/off" would not make it clear which temperature environment (boiler, outside temperature?) would be meant. Within the proper category, we could then look for relatively "safe" parameter numbers and mark the parameter as one that would need to be shifted in case the parameter number is claimed by an "official" parameter.
